.circleci/config.yml

```diff
@@ -136,391 +145,396 @@ jobs:
-            brew install pyenv
-
-      - restore_cache:
-          keys:
-            - pyenv-{{ .Environment.CIRCLE_JOB }}-xcode12.2.0
-
+            python -m venv env
+            . env/bin/activate
+            pip install -r requirements.txt
+            pip install -r docs/requirements.txt
+      - run: *unix-build
       - run:
-          name: install python
+          name: build docs
           command: |
-            pyenv install << parameters.python-version>> -s
-
-      - save_cache:
-          paths:
-            - ~/.pyenv
-          key: pyenv-{{ .Environment.CIRCLE_JOB }}-xcode12.2.0
-
-      - run: *osx-install-template
-
+            . env/bin/activate
+            make -C docs/ cpp html
+      - store_artifacts:
+          path: ./docs/build/html
       - run:
-          name: create bdist_wheel
+          name: doctest
           command: |
             . env/bin/activate
-            python setup.py bdist_wheel
-
+            make -C docs/ doctest
       - run:
-          name: deploy
+          name: linkcheck
           command: |
             . env/bin/activate
-            python -m pip install twine
-            twine upload -u $PYPI_USERNAME -p $PYPI_PASSWORD --skip-existing ./dist/*
-
-  #
-  # win
-  #
+            make -C docs/ linkcheck
 
-  test-win:
+  test-linux:
     parameters:
       python-version:
         type: string
-      architecture:
+      numpy-version:
+        type: string
+      dimod-version:
         type: string
 
-    executor:
-      name: win/default
-
-    environment:
-      CL: /d2FH4-
+    docker:
+      - image: circleci/python:<< parameters.python-version >>
 
     steps:
       - checkout
-
-      - run: &win-install-python-template
-          name: install python and create virtualenv
+      - attach_workspace:
+          at: dist
+      - run:
+          name: install
           command: |
-            nuget install python<< parameters.architecture >> -Version << parameters.python-version >> -ExcludeVersion -OutputDirectory .
-            .\python<< parameters.architecture >>\tools\python.exe --version
-            .\python<< parameters.architecture >>\tools\python.exe -m pip install virtualenv
-            .\python<< parameters.architecture >>\tools\python.exe -m virtualenv env
-
-      - run: &win-install-dependencies-template
-          name: install dependencies
+            python -m venv env
+            . env/bin/activate
+            pip install 'dimod<< parameters.dimod-version >>' 'numpy<< parameters.numpy-version >>' --only-binary=dimod,numpy
+            pip install dwave-preprocessing --no-index -f dist/ --no-deps --force-reinstall
+      - run: &unix-run-tests
+          name: run tests
           command: |
-            env\Scripts\activate.ps1
-            python --version
+            . env/bin/activate
+            cd tests/
             pip install -r requirements.txt
-            pip install -r tests\requirements.txt
+            python -m unittest
```

Comment:
What about:
```
pip install -r tests/requirements.txt
python -m unittest discover
```

Reply:
We need to run them within `tests/`, otherwise it will use the local file structure rather than the installed version  (there are other ways to get around that, but this is IMO the easiest).

Reply:
Ah that makes sense, thanks